tests/test_dataserver.py

```diff
@@ -1,8 +1,32 @@
+from pathlib import Path
 from hifz.dataserver import DataServer
 from hifz.models import Card
 
+def test_dataserver_returns_card_csv():
+    server = DataServer()
+    temp_file_path = "data/test_arabic.csv"
+    with Path(temp_file_path).open() as f:
+        f.writelines([
+            "front,back",
+            "ا,alif (a as in bad or A as in fAther)",
+            "ب,baa'"
+            ])
+    cards = server.read_entries(temp_file_path)
+    Path(temp_file_path).unlink()
+    assert Card("ب", "baa'") in cards
 
-def test_dataserver_returns_card():
+def test_dataserver_returns_card_json():
     server = DataServer()
-    cards = server.read_entries("data/arabic_letters.csv")
+    temp_file_path = "data/test_arabic.json"
+    with Path(temp_file_path).open() as f:
+        f.write(str([
+        {
+        "front": "ا",
+        "back": "alif (a as in bad or A as in fAther)"
+        },{
+            "front": "ب",
+            "back": "baa"
+        }]))
+    cards = server.read_entries(temp_file_path)
+    Path(temp_file_path).unlink()
```

Comment:
I believe what this code does is 

1. Overwrite the dataset in data/
2. Read the newly overwritten data from data/
3. Delete the dataset from data/ (unlink)
4. Test functionality

I'm currently at six flags so I can't test this myself, but this seems like unintentional behavior. I think having the example datasets in data/ is useful, so we shouldn't overwrite and delete them. Please let me know if I'm not understanding this code correctly.

Reply:
I just changed it to use tmp_file but the logic was:
1. create a new file
2. save test data to that file
3. save the card set from read_entries as cards
4. delete the file
5. assert that the card set is correct

Did I have any errors in that?